fapolicy_analyzer/ui/features/daemon_feature.py

```diff
@@ -0,0 +1,179 @@
+    received_daemon_stop,
+)
+from fapolicy_analyzer.ui.reducers import daemon_reducer
+from gi.repository import GLib
+from redux import (
+    Action,
+    combine_epics,
+    create_feature_module,
+    of_init_feature,
+    of_type,
+    ReduxFeatureModule,
+)
+from fapolicy_analyzer.ui.strings import DAEMON_INITIALIZATION_ERROR
+from rx import of, pipe
+from rx.operators import catch, map
+from typing import Callable
+
+DAEMON_FEATURE = "daemon"
+_fapd_status: bool = False
+_fapd_ref: Handle = None
+
+
+def create_daemon_feature(dispatch: Callable, daemon=None) -> ReduxFeatureModule:
+    """
+    Creates a Redux feature of type Daemon
+
+    Keyword arguments:
+    dispatch -- the Redux Store dispatch method, used for dispatching actions
+    daemon -- the fapolicy_analyzer.daemon object, defaults to None. If not
+    provided, a new Daemon object will be initialized.  Used for testing
+    purposes only.
+    """
+    def _init_daemon() -> Action:
+        logging.debug("_init_daemon() -> Action")
+
+        def acquire_daemon():
+            logging.debug("acquire_daemon()")
+            try:
+                daemon = Handle("fapolicyd")
+                GLib.idle_add(finish, daemon)
+            except RuntimeError:
+                logging.exception(DAEMON_INITIALIZATION_ERROR)
+                GLib.idle_add(sys.exit, 1)
+
+        def monitor_daemon(timeout=5):
+            global _fapd_ref, _fapd_status
+            while True:
+                bStatus = is_fapolicyd_active()
+                if(bStatus != _fapd_status):
+                    logging.debug("monitor_daemon():Dispatching update request")
+                    _fapd_status = bStatus
+                    dispatch(received_daemon_status_update(bStatus))
+                sleep(timeout)
+
+        def start_daemon_monitor():
+            logging.debug("start_daemon_monitor()")
+            thread = Thread(target=monitor_daemon)
+            thread.daemon = True
+            thread.start()
+
+        def finish(daemon: Handle):
+            global _fapd_ref, _fapd_status
+            _fapd_ref = daemon
+            _fapd_status = is_fapolicyd_active()
+
+        if daemon:
+            finish(daemon)
+        else:
+            acquire_daemon()
+            if "NO_DAEMON_MONITORING" not in os.environ:
+                start_daemon_monitor()
+        return init_daemon()
+
+    def _daemon_reload(action: Action) -> Action:
+        logging.debug("_daemon_reload(action: Action) -> Action")
+        return received_daemon_reload()
+
+    def _daemon_start(action: Action) -> Action:
+        logging.debug("_daemon_start(action: Action) -> Action")
+        start_fapolicyd()
+        return received_daemon_start()
+
+    def _daemon_status(action: Action) -> Action:
+        logging.debug(f"_daemon_status(action: {action}) -> Action")
+        status = is_fapolicyd_active()
+        logging.debug(f"_daemon_status::Fapolicyd status: {status}")
+        _fapd_status = status
```

Comment:
I don't see a reason to assign a separate variable here just to use for logging.  I'd suggest: 
```suggestion
        _fapd_status = is_fapolicyd_active()
        logging.debug(f"_daemon_status::Fapolicyd status: {_fapd_status}")
```
Same for the `_daemon_status_update` method below.

Reply:
Just part of the debugging process/noise. I expect to remove a lot of the logging msgs in the reducer/feature modules.

Reply:
Cleaned up.